### PR TITLE
EVP: Make the KEYEXCH implementation leaner

### PR DESCRIPTION
**Pull request by levitte**
Because the algorithm to use is decided already when creating an
EVP_PKEY_CTX regardless of how it was created, it turns out that it's
unnecessary to provide the KEYEXCH method explicitly, and rather
always have it be fetched implicitly.

This means fewer changes for applications that want to use new key
exchange algorithms / implementations.

### CODE REVIEW

**Comment by levitte** [comment]
This turned out to be much much easier than #10303...  possibly because I had already put in the work in #10303 and this practically became just a copy-n-paste job

**Comment by mattcaswell** [comment]
I'll hold back reviewing this one for a bit until #10303 is a little further along. Most of the same comments from there apply here too.

**Comment by levitte** [comment]
Yup

**Comment by levitte** [comment]
I hope this is complete enough now

**Comment by levitte** [comment]
Ping

**Comment by levitte** [comment]
This will not build right before #10308 is merged

**Comment by levitte** [comment]
#10308 is now in, and this is now rebased

File: doc/man3/EVP_PKEY_derive.pod
Comment (mattcaswell) [inline doc/man3/EVP_PKEY_derive.pod]: Either `The EVP_PKEY_derive() function derives...` or just `EVP_PKEY_derive() derives...` - but not as it is at the moment (an extra "The").

**Review by mattcaswell — APPROVED** [review]
Approved assuming the documentation nit is fixed.

File: doc/man3/EVP_PKEY_derive.pod
Comment (levitte) [reply doc/man3/EVP_PKEY_derive.pod]: You're right. I will fix that

**Comment by levitte** [comment]
Merged.

c0e0984f125ee30820520829c4c112b1e166f871 EVP: Make the KEYEXCH implementation leaner

**Pull request closed without merging**
